File: laboratorioMombelli/geometria/triangolo.py
from punto import Punto
from segmento import Segmento
from math import sqrt, pow

class Triangolo:

    def __init__(self, A: Punto, B: Punto, C: Punto):
        self.A = A
        self.B = B
        self.C = C

        self.AB = Segmento(A, B)
        self.AC = Segmento(A, C)
        self.BC = Segmento(B, C) 
    
    def perimetro(self):
        return self.AB.lunghezza() + self.AC.lunghezza() + self.BC.lunghezza()
    
    def area(self):
        sp = self.perimetro() / 2
        return sqrt(sp * (sp - self.AB.lunghezza()) * (sp - self.AC.lunghezza()) * (sp - self.BC.lunghezza()))
    
    def __str__(self):
        return f" Trinagolo con perimetro {self.perimetro()} e superfice {self.area()}"

File: laboratorioMombelli/Java/eclipse-workspace/Prj05_GeometriaPiana/src/figure/Triangolo.java
package figure;

public class Triangolo extends LuogoGeometrico implements Contornabile {
	
	Punto a, b, c;
	Segmento ab, bc, ac;
	
	public Triangolo(Punto a, Punto b, Punto c) {
		this.nome = "Triangolo";
		this.a = a;
		this.b = b;
		this.c = c;
		
		this.ab = new Segmento(a,b);
		this.ac = new Segmento(a,c);
		this.bc = new Segmento(b,c);
		
	}
	
	
	public double perimetro() {
		return ab.lunghezza() + ac.lunghezza() + bc.lunghezza();
	}
	
	public double area() {
		double area = 0;
		double sp = perimetro() / 2;
		
		area = Math.sqrt(
				sp *
				(sp - ab.lunghezza()) *
				(sp - ac.lunghezza()) *
				(sp - bc.lunghezza())
				);
		return area;
	}


	@Override
	public String toString() {
		return "Triangolo [perimetro()=" + perimetro() + ", area()=" + area() + "]";
	}


	@Override
	public double contorno() {
		return perimetro();
	}
	
	

}
